### ISSUE TITLE
Nested hidden form fields causes JavaScript error in the CMS

### ISSUE THREAD

**Issue by UndefinedOffset** [silverstripe/silverstripe-framework#5618]
It looks like when you have nested form fields that are not visible a script error is thrown form LeftAndMain.EditForm.js line 350 (in 3.3's version). In short when restoreFieldFocus() tries to find the next visible element it's assuming that the next visible element is the next parent ``.field`` which may not be true in the case of say a FieldGroup inside of a ToggleComposite which is collapsed.

It looks like this issue can occur in [3.2](https://github.com/silverstripe/silverstripe-framework/blob/3.2/admin/javascript/LeftAndMain.EditForm.js#L321), [3.3](https://github.com/silverstripe/silverstripe-framework/blob/3.3/admin/javascript/LeftAndMain.EditForm.js#L350) and [3.4](https://github.com/silverstripe/silverstripe-framework/blob/3.4/admin/javascript/LeftAndMain.EditForm.js#L350).

I think the fix is as simple as adding a ``:visible`` flag to the selector where the nearest visible element is looked up. The diff below was made against 3.3.2, I can open a pull if the change makes sense.

```diff
diff --git a/admin/javascript/LeftAndMain.EditForm.js b/admin/javascript/LeftAndMain.EditForm.js
index a9ee20f..983c18f 100644
--- a/admin/javascript/LeftAndMain.EditForm.js
+++ b/admin/javascript/LeftAndMain.EditForm.js
@@ -318,7 +318,7 @@
 
                    //Fall back to nearest visible element if hidden (for select type fields)
                    if(!$(elementID).is(':visible')){
-                       elementID = '#' + $(elementID).closest('.field').attr('id');
+                       elementID = '#' + $(elementID).closest('.field:visible').attr('id');
                        scrollY = $(elementID).position().top;
                    }
```

**Comment by dhensby** [comment]
That is the craziest way to get the element.

Can't that be condensed to to:

```
if(!$(elementID).is(':visible')){
    scrollY = $(elementID).closest('.field:visible').position().top;
}
```

Or is `elementID` needed later?

**Comment by dhensby** [comment]
PS: looks like a good fix, can we get a PR against 3.2 please?

**Comment by UndefinedOffset** [comment]
ya I thought it was a bit odd looking ;) lol will open a pull request shortly to address the primary issue. It looks like the elementID is actually used later to focus the element.

**Comment by dhensby** [comment]
h'ok - thanks

**Comment by UndefinedOffset** [comment]
there, opened a pull to 3.2... initially had it set to master so i closed that and re-opened (oops)

**Comment by dhensby** [comment]
thanks!

**Comment by UndefinedOffset** [comment]
Will this make 3.4.0? or will it need to wait until 3.4.1? Just trying to gauge how to set an expectation internally. I'm assuming 3.3.3 is several weeks out yet.

**Comment by UndefinedOffset** [comment]
@dhensby FYI this impacts 3.3 branch and 3.4 branch as well.

**Comment by sminnee** [comment]
We don't generally do merge-forwards of `3.2` to the `3.4-rc1` branch so I'd expect that this will be released in `3.4.1`, `3.3.3`, and `3.2.5`.

If you need it sooner, you may want to try https://github.com/cweagans/composer-patches
You can use https://github.com/silverstripe/silverstripe-framework/commit/0dcffda0e82eac7b8b54c1dffd078ec4486490cf.patch as the patch URL to grab it straight from the source.
